### PR TITLE
Fix dh 150 create form falied validation text

### PR DESCRIPTION
**Pull request by kiddhustle**
fixes following:

* having to refill fields on validation error
* submission error on client_relationship_manager fields after validation error

### CODE REVIEW

File: src/forms/createinvestment.js [hunk 1]
```diff
@@ -107,6 +107,22 @@ function sectorHasSubs (id) {
...
+   * add the selected class to a checked fields label (for UI, do it here rather than in Nunjucks template)
+   */
+  // console.log('updateSelectedStateDisplay')
+  let aRadioBtns = [].slice.call(document.querySelectorAll('input[type=radio][checked]'))
+  aRadioBtns.forEach((input, i) => {
+    let dParent = input.closest('label')
+    if (dParent) {
+      dParent.classList.add('selected')
+      // console.log('dParent')
+      // console.log(dParent)
```
Comment (jim68000) [inline src/forms/createinvestment.js]: These need zapping

File: src/forms/createinvestment.js [hunk 2]
```diff
@@ -107,6 +107,22 @@ function sectorHasSubs (id) {
...
 }
 
+function updateSelectedStateDisplay () {
+  /**
+   * add the selected class to a checked fields label (for UI, do it here rather than in Nunjucks template)
+   */
+  // console.log('updateSelectedStateDisplay')
+  let aRadioBtns = [].slice.call(document.querySelectorAll('input[type=radio][checked]'))
+  aRadioBtns.forEach((input, i) => {
+    let dParent = input.closest('label')
```
Comment (jim68000) [inline src/forms/createinvestment.js]: Does this get transpiled into something sensible? .closest is not supported on any IE iirc

File: src/controllers/investmentcontroller.js [hunk 3]
```diff
@@ -183,7 +185,20 @@ function postProject (req, res) {
   if (errors) {
     genCSRF(req, res)
     res.locals.errors = errors
-    return create(req, res)
+    req.session.userInput = req.body
+    // account for multiple client_relationship_manager entries
+    if (Array.isArray(req.session.userInput.client_relationship_manager)) {
+      let parsedClientManager = req.session.userInput.client_relationship_manager.find(val => val !== '')
+      req.session.userInput.client_relationship_manager = parsedClientManager || ''
```
Comment (jim68000) [inline src/controllers/investmentcontroller.js]: This is a good point. I have CRM as singular in the API design, but it is actually an array.

File: src/lib/utils.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const copyObj = function (obj) {
+  return JSON.parse(JSON.stringify((obj)))
```
Comment (jim68000) [inline src/lib/utils.js]: @ztolley Zac, do you have a view on this? I know you use Object.assign to achieve the same thing, but that runs into problems with object references. 

TBH, this looks hacky, but I know it's the recommended way to do deep cloning...

Comment (ztolley) [inline src/lib/utils.js]: Object.assign should work and not have problems, the idea is it creates a new object (nested) so it's a copy and not a reference which is what you are trying to do here.

File: src/views/investment/summary.html [hunk 5]
```diff
@@ -1,6 +1,11 @@
 {% import "macros/trade.html" as trade %}
-  {% import "./investment_macros.html" as local %}
+{% import "./investment_macros.html" as local %}
+{% macro input_attr(name, val) %}{% if val %}{{name}}="{{name}}"{% endif %}{% endmacro %}
 
+{#<textarea>{{userInput|dump}}</textarea>
```
Comment (ztolley) [inline src/views/investment/summary.html]: If this isn't used can it be deleted?

File: src/views/investment/summary.html [hunk 6]
```diff
@@ -33,95 +38,95 @@ <h2 class="heading-medium">Additional company</h2>
...
     class="select-autocomplete-js")
     }}
+    {% set amcrm_checked = true if userInput.amcr == 'yes' else false %}
+    {{ local.inlineradio("amcrm", "client-rel-reveal", "client-rev-reveal", label='Are you the Client Relationship Manager for this project?',
+      checked=amcrm_checked) }}
 
-    {{ local.inlineradio("amcrm", "client-rel-reveal", "client-rev-reveal", label='Are you the Client Relationship Manager for this project?') }}
-
-    {{ trade.textbox("client_relationship_manager", label="Client Relationship Manager", class="hidden panel", id="client_relationship_manager") }}
+    {# Two elements with same name, this will probably cause problems #}
```
Comment (ztolley) [inline src/views/investment/summary.html]: so does this need to be fixed?

**Comment by kiddhustle** [comment]
console.logs zapped

**Comment by kiddhustle** [comment]
is this PR good to go now then?

File: src/lib/utils.js (hunk 4 above)
Comment (kiddhustle) [inline src/lib/utils.js]: @jim68000 this good to go now?

Comment (jim68000) [inline src/lib/utils.js]: Yup. Merge

**Pull request merged**